addons/stock/report/stock_forecasted.py

```diff
@@ -117,26 +129,54 @@ def _prepare_report_line(self, quantity, move_out=None, move_in=None, replenishm
         move_in_id = move_in.id if move_in else None
         move_out_id = move_out.id if move_out else None
 
-        return {
-            'document_in': move_in._get_source_document() if move_in else False,
-            'document_out': move_out._get_source_document() if move_out else False,
+        line = {
+            'move_in' : False,
+            'move_out' : False,
+            'document_in' : False,
+            'document_out' : False,
+            'receipt_date': False,
+            'delivery_date': False,
             'product': {
                 'id': product.id,
-                'display_name': product.display_name
+                'display_name': product.display_name,
             },
             'replenishment_filled': replenishment_filled,
-            'uom_id': product.uom_id,
-            'receipt_date': format_date(self.env, move_in.date) if move_in else False,
-            'delivery_date': format_date(self.env, move_out.date) if move_out else False,
             'is_late': is_late,
             'quantity': float_round(quantity, precision_rounding=product.uom_id.rounding),
-            'move_out': move_out,
-            'move_in': move_in,
             'reservation': reservation,
             'is_matched': any(move_id in [move_in_id, move_out_id] for move_id in move_to_match_ids),
+            'uom_id' : product.uom_id.read()[0],
         }
+        if move_in:
+            document_in = move_in._get_source_document()
+            line.update({
+                'move_in' : move_in.read()[0],
+                'document_in' : {
+                    '_name' : document_in._name,
+                    'id' : document_in.id,
+                    'name' : document_in.name,
```

Comment:
You could use display_name. It uses the name_get and allow to override document name and add extra information. It's usually better to use display_name rather than name